src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,122 @@
+        return elevatorEncoder.getSelectedSensorPosition();
+    }
+
+    private double getAngleMotorVelocity() {
+        return angleEncoder.getVelocity().getValue();
+    }
+
+    private double getElevatorMotorVelocity() {
+        return elevatorEncoder.getSelectedSensorVelocity();
+    }
+
+    private double getAngleMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastAngleProfileGeneration;
+    }
+
+    private double getElevatorMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastElevatorProfileGeneration;
+    }
+
+    private void generateAngleMotorProfile(Rotation2d targetAngle) {
+        angleMotorProfile = new TrapezoidProfile(
+                ArmConstants.ANGLE_CONSTRAINS,
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getAngleMotorPositionDegrees(), getElevatorMotorVelocity())
+        );
+        lastAngleProfileGeneration = Timer.getFPGATimestamp();
+    }
+
+    private void generateElevatorMotorProfile(Rotation2d targetAngle) {
+        elevatorMotorProfile = new TrapezoidProfile(
+                ArmConstants.ELEVATOR_CONSTRAINS,
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getElevatorMotorPositionDegrees(), getAngleMotorVelocity())
+        );
+        lastElevatorProfileGeneration = Timer.getFPGATimestamp();
+    }
+
+    private void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            masterAngleMotor.stopMotor();
+            followerAngleMotor.stopMotor();
+            return;
+        }
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleMotorProfileTime());
+        double pidOutput = ArmConstants.ANGLE_PID_CONTROLLER.calculate(
+                getAngleMotorPositionDegrees(),
+                targetState.position
+        );
+        double feedForward = ArmConstants.ANGLE_MOTOR_FEEDFORWARD.calculate(
+                Units.degreesToRadians(targetState.position),
+                Units.degreesToRadians(targetState.velocity)
+        );
+        masterAngleMotor.setVoltage(pidOutput + feedForward);
+        followerAngleMotor.setVoltage(pidOutput + feedForward);
+    }
+
+    private void setTargetElevatorFromProfile() {
+        if (elevatorMotorProfile == null) {
+            masterElevatorMotor.stopMotor();
+            followerElevatorMotor.stopMotor();
+            return;
+        }
+        TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorMotorProfileTime());
+        double pidOutput = ArmConstants.ELEVATOR_PID_CONTROLLER.calculate(
+                getElevatorMotorPositionDegrees(),
+                targetState.position
+        );
+        double feedForward = ArmConstants.ELEVATOR_MOTOR_FEEDFORWARD.calculate(
+                Units.degreesToRadians(targetState.position),
+                Units.degreesToRadians(targetState.velocity)
```

Comment:
You have a logic problem here. Look at what params this accepts, and do you want to input radians?